### ISSUE TITLE
3.2.0 regression

### ISSUE THREAD

**Issue by steida** [pburtchaell/redux-promise-middleware#90]
Catch on promise returned from dispatch doesn't work anymore. Instead, the error is passed into then handler.

**Comment by pburtchaell** [comment]
@steida Can you provide more details and an example of how to reproduce?

Perhaps I am misunderstanding your issue, but [we test to ensure errors can be handled with `catch`](https://github.com/pburtchaell/redux-promise-middleware/blob/master/test/index.spec.js#L330-L339).

**Comment by steida** [comment]
I will try to figure out. Btw, `Promise.reject(null)` is an anti-pattern. http://bluebirdjs.com/docs/warning-explanations.html#warning-a-promise-was-rejected-with-a-non-error

**Comment by steida** [comment]
So I cloned project and it seems a lot of your tests simply doesn't work. You are using await wrong.

**Comment by steida** [comment]
```
await actionDispatched.catch(({ action, reason }) => {
  console.log('fffffffffffffffff'); // not called
  expect(action).to.eql({
    type: `${nullRejectAction.type}_REJECTED`,
    error: true
  });
});
```

**Comment by leonfs** [comment]
I have been experiencing the same issue. On Promise Rejection can't catch the error.

**Comment by pburtchaell** [comment]
Gotcha, @steida. I'll take a look this evening.

**Comment by steida** [comment]
@pburtchaell Remember RED GREEN BLUE TDD rules. At first write FAILING test, then fix that test, then refactor. You can't trust test which never failed.

**Comment by steida** [comment]
@pburtchaell In Mocha there is done callback for such testing.

**Comment by pburtchaell** [comment]
Yeah, `async...await` should also work. I need to just fix how I am using the `await` statement. The problem was when we moved from v2 to v3, I didn't update the tests correctly — my fault.

**Comment by pburtchaell** [comment]
This is patched in `redux-promise-middleware@3.3.0`.
